### ISSUE TITLE
unnecessaryIsInstance not reported for bytes type on bytearray | bytes when disableBytesTypePromotions = false

### ISSUE THREAD

**Issue by ItsDrike** [DetachHead/basedpyright#495]
For some reason, when `disableBytesTypePromotions = false`, basedpyright no longer reports `unnecessaryIsInstance` in the code below:
```python
x: bytes = ...  # pyright: ignore[reportAssignmentType]

if isinstance(x, (bytearray, bytes)):  # problematic line
    ...
```

Yet with the `disableBytesTypePromotions  = true`, the error is reported properly.

The `disableBytesTypePromotions` just controles whether or not `bytes` are considered a supertype of `bytearray` and `memoryview`, however, this setting shouldn't matter here, as `bytes` will always be a subtype of `bytes | bytearray`, the check should be unnecessary in both cases (unless I'm missing something).

**Comment by DetachHead** [comment]
i guess whatever special casing pyright does to support this legacy behavior doesn't work properly with `reportUnnecessaryIsInstance`. tbh i would consider fixing this to be low priority considering [this functionality is deprecated](https://peps.python.org/pep-0688/#no-special-meaning-for-bytes) and will probably be removed from pyright eventually.

on a related note, i also intend to add an option to disable other type promotions (#319) because i think it's a really dumb feature in general
